### PR TITLE
Hpcc 14563 Support for custom transaction id formats

### PR DESCRIPTION
**Pull request by rpastrana**
Fetch custom trxid from logging manager, if available.

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review  	76bfa24

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 1]
```diff
@@ -522,6 +525,38 @@ void EsdlServiceImpl::handleServiceRequest(IEspContext &context,
...
+            trxidbasics.setValue("TransactionMethod", mthName);
+            trxidbasics.setValue("TransactionESPIP", uniqueId.str()); // this should be unique ID
+
+            StringBuffer trxidstatus;
+            if (!m_oLoggingManager->getTransactionID(&trxidbasics,trxid, trxidstatus))
+                ESPLOG(LogMin,"DESDL: Logging Agent generated Transaction ID failed: %s", trxidstatus.str());
+        }
+        else
+            ESPLOG(LogNormal,"DESDL: Transaction ID could not be fetched from logging manager!");
+       }
```
Comment (wangkx) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: wrong indent

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 2]
```diff
@@ -2469,6 +2504,7 @@ int EsdlBindingImpl::onGetRoxieBuilder(CHttpRequest* request, CHttpResponse* res
 
     StringBuffer roxiemsg, serviceQName, methodQName;
     IEspContext * context = request->queryContext();
+    context->setTransactionID("ROXIETEST-NOTRXID");
```
Comment (wangkx) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: test code?

File: esp/services/esdl_svc_engine/esdl_svc_engine.cpp [hunk 3]
```diff
@@ -41,28 +41,24 @@ CEsdlSvcEngineSoapBindingEx::CEsdlSvcEngineSoapBindingEx(IPropertyTree* cfg, con
...
 {
-    const char *querytype = tgtcfg->queryProp("@querytype");
-    if (!querytype || !strieq(querytype, "ROXIE")) //only roxie?
-        return NULL;
-
-    StringBuffer trxid;
-    generateTransactionId(context, trxid);
+    //const char *querytype = tgtcfg->queryProp("@querytype");
+    //if (!querytype || !strieq(querytype, "ROXIE")) //only roxie?
+    //    return NULL;
```
Comment (wangkx) [inline esp/services/esdl_svc_engine/esdl_svc_engine.cpp]: Remove those commented out code?

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 4]
```diff
@@ -522,6 +525,38 @@ void EsdlServiceImpl::handleServiceRequest(IEspContext &context,
...
+            short int port;
+            context.getServAddress(wsaddress, port);
+            VStringBuffer uniqueId("%s:%d-%u", wsaddress.str(), port, (unsigned) (memsize_t) GetCurrentThreadId());
+
+            StringAttrMapping trxidbasics;
+            StringBuffer creationTime;
+            creationTime.setf("%u", context.queryCreationTime());
+            trxidbasics.setValue("TransactionDateTime", creationTime.str());
+            trxidbasics.setValue("TransactionMethod", mthName);
+            trxidbasics.setValue("TransactionESPIP", uniqueId.str()); // this should be unique ID
```
Comment (wangkx) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: Is it possible to use the #define's in loggingagentbase.hpp to replace those hard coded strings?

**Comment by wangkx** [comment]
@rpastrana a few comments

File: esp/services/esdl_svc_engine/esdl_binding.cpp (hunk 4 above)
Comment (rpastrana) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: I was going to comment that those those should actually be public static const member variables in the base logagent, or an enumeration.

File: esp/services/esdl_svc_engine/esdl_binding.cpp (hunk 2 above)
Comment (rpastrana) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: this is for Roxie Test, we don't generate an actual trxid

**Comment by rpastrana** [comment]
Closing in favor of: https://github.com/hpcc-systems/HPCC-Platform/pull/8953

**Pull request closed without merging**